tests/xharness/XmlResultParser.cs

```diff
@@ -0,0 +1,267 @@
+						total += long.Parse (reader ["total"]);
+						errors += long.Parse (reader ["errors"]);
+						failed += long.Parse (reader ["failed"]);
+						skipped += long.Parse (reader ["skipped"]);
+					}
+					if (reader.NodeType == XmlNodeType.Element && reader.Name == "collection") {
+						var testCaseName = reader ["name"].Replace ("Test collection for ", "");
+						writer.WriteLine (testCaseName);
+						var time = reader.GetAttribute ("time") ?? "0"; // some nodes might not have the time :/
+												// get the first node and then move in the siblings of the same type
+						reader.ReadToDescendant ("test");
+						do {
+							if (reader.Name != "test")
+								break;
+							// read the test cases in the current node
+							var status = reader ["result"];
+							switch (status) {
+							case "Pass":
+								writer.Write ("\t[PASS] ");
+								break;
+							case "Skip":
+								writer.Write ("\t[IGNORED] ");
+								break;
+							case "Fail":
+								writer.Write ("\t[FAIL] ");
+								break;
+							default:
+								writer.Write ("\t[FAIL] ");
+								break;
+							}
+							writer.Write (reader ["name"]);
+							if (status == "Fail") { //  we need to print the message
+								reader.ReadToDescendant ("message");
+								writer.Write ($" : {reader.ReadElementContentAsString ()}");
+								reader.ReadToNextSibling ("stack-trace");
+								writer.Write ($" : {reader.ReadElementContentAsString ()}");
+							}
+							// add a new line
+							writer.WriteLine ();
+						} while (reader.ReadToNextSibling ("test"));
+						writer.WriteLine ($"{testCaseName} {time} ms");
+					}
+				}
+			}
+			var passed = total - errors - failed - notRun - inconclusive - ignored - skipped - invalid;
+			string resultLine = $"Tests run: {total} Passed: {passed} Inconclusive: {inconclusive} Failed: {failed + errors} Ignored: {ignored + skipped + invalid}";
+			writer.WriteLine (resultLine);
+
+			return (resultLine, total == 0 | errors != 0 || failed != 0);
+		}
+
+		public static string GetXmlFilePath (string path, XmlResultType xmlType)
+		{
+
+			using (var streamReaderTmp = new StreamReader (path)) {
+				var fileName = Path.GetFileName (path);
+				var newFilename = "";
+				switch (xmlType) {
+				case XmlResultType.TouchUnit:
+				case XmlResultType.NUnit:
+					newFilename = path.Replace (fileName, $"nunit-{fileName}");
+					break;
+				case XmlResultType.xUnit:
+					newFilename = path.Replace (fileName, $"xunit-{fileName}");
+					break;
+				}
+				return newFilename;
+			}
+		}
+
+		public static void CleanXml (string source, string destination)
+		{
+			using (var reader = new StreamReader (source))
+			using (var writer = new StreamWriter (destination)) {
+				string line;
+				while ((line = reader.ReadLine ()) != null) {
+					if (line.StartsWith ("ping", StringComparison.InvariantCulture) || line.Contains ("TouchUnitTestRun") || line.Contains ("NUnitOutput") || line.Contains ("<!--")) {
```

Comment:
`StringComparison.InvariantCulture` is much more costly than `Ordinal`

Reply:
unresolving - that was not fixed!

please don't resolve other comments, i.e. let the commenter resolve
otherwise there's a risk that the comment will not be addressed and missed